packages/near-membrane-base/src/membrane.ts

```diff
@@ -2702,18 +2850,21 @@ export function createMembraneMarshall(
                           proxy,
                           shadowTarget,
                       } = this;
-                      const possibleIndex = typeof key === 'string' ? +key : -1;
+                      let useFastPath = useFastPathForTypedArrays;
+                      if (!useFastPath && typeof key === 'string') {
+                          const possibleIndex = +key;
+                          useFastPath =
+                              possibleIndex > -1 &&
+                              possibleIndex < foreignTargetTypedArrayLength &&
+                              NumberIsInteger(possibleIndex);
```

Comment:
Because typed arrays are dense and indexes are own values we continue to use the fast path for those accesses.